client/cli/src/main.rs

```diff
@@ -524,11 +525,21 @@ async fn handle_command(
             CliErrorKind::GeneralFederationError,
             "failed to execute spend (no further information)",
         ),
-        Command::Fetch => match client.fetch_all_notes().await {
-            Ok(result) => Ok(CliOutput::Fetch { issuance: (result) }),
+        Command::Fetch => match client.claim_incoming_contracts(&mut rng).await {
+            Ok(contracts) => match client.fetch_all_notes().await {
+                Ok(result) => Ok(CliOutput::Fetch {
+                    claimed_incoming_contracts: contracts,
+                    issuance: (result),
+                }),
+                Err(error) => Err(CliError::from(
+                    CliErrorKind::GeneralFederationError,
+                    "failed to fetch notes",
+                    Some(Box::new(error)),
+                )),
+            },
```

Comment:
Nesting `match`es isn't nice, consider putting the fetch code into a function that resturns a `Result` and then using `map_err` and the `?` operator instead.

Reply:
OK sure, will give that a try thanks.

Reply:
I've removed the `match` expressions now. Happy to refactor further if required.